fs: Fix MKFS Kconfig preventing documentation

decsny opened the pull request:
This KConfig  was preventing documentation of MKFS from being included by doxygen.

de-nordic reviewed (changes requested):
OK for the include/zephyr/fs/fs.h, no for the others.
Users that do not use mkfs on devices in the field should not be forced to waste flash on this feature.

Commit:
include: fs: Remove conditional around MKFS

The MKFS function declaration is conditionally compiled
based on the MKFS kconfig, which prevents it from being documented.
Remove the conditional.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented:
@de-nordic updated
